lib/logical_solver/presolve.cpp

```diff
@@ -0,0 +1,889 @@
+#include "presolve.hpp"
+#include <iostream>
+
```

Comment:
need `#include <algorithm>` for `std::find()` function (doesn't compile for me without it).

Reply:
Should be included now.